polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1753,27 +1757,27 @@ impl<Config: config::Config> XcmExecutor<Config> {
 	) -> Result<(), XcmError> {
 		let mut failed_deposits = Vec::with_capacity(to_deposit.len());
 
-		let mut deposit_result = Ok(());
 		for asset in to_deposit.assets_iter() {
-			deposit_result = Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context);
-			// if deposit failed for asset, mark it for retry after depositing the others.
-			if deposit_result.is_err() {
+			// if deposit failed for asset, mark it for retry.
+			if Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context).is_err() {
 				failed_deposits.push(asset);
 			}
 		}
-		if failed_deposits.len() == to_deposit.len() {
-			tracing::debug!(
-				target: "xcm::execute",
-				?deposit_result,
-				"Deposit for each asset failed, returning the last error as there is no point in retrying any of them",
-			);
-			return deposit_result;
-		}
-		tracing::trace!(target: "xcm::execute", ?failed_deposits, "Deposits to retry");
+
+		tracing::trace!(
+			target: "xcm::execute",
+			?failed_deposits,
+			"First‐pass failures, about to retry"
+		);
 
 		// retry previously failed deposits, this time short-circuiting on any error.
 		for asset in failed_deposits {
-			Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context)?;
+			if let Err(e) = Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context) {
+				// Ignore dust deposit errors.
+				if !matches!(e, XcmError::FailedToTransactAsset(s) if *s == *"BelowMinimum") {
```

Comment:
I would prefer to use enum directly `sp_runtime::TokenError::BelowMinimum.into()` instead of hard-coded string,
TokeError has `impl From<TokenError> for &'static str {` and `sp_runtime` is already here as dep:

So something like this:
```
				if !matches!(e, XcmError::FailedToTransactAsset(s) if *s == sp_runtime::TokenError::BelowMinimum.into()) {
```
or 
```
				if e != XcmError::FailedToTransactAsset(sp_runtime::TokenError::BelowMinimum.into()) {
```